packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useLocationsData } from '../../../context/actions';
+import { LocationAccess } from '../../../context/types';
+import { compareStrings } from '../../../context/controls/Table';
+import { ButtonElement, IconElement } from '../../../context/elements';
+import { capitalize } from '@aws-amplify/ui';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+const getCompareFn = (selection: string) => {
+  switch (selection) {
+    case 'scope':
+    case 'type':
+    case 'permission':
+      return compareStrings;
+  }
+};
+
+const getColumnItem = ({
+  key,
+  selection,
+  direction,
+  onTableHeaderClick,
+}: {
+  key: string;
+  selection: string;
+  direction: SortDirection;
+  onTableHeaderClick: (location: string) => void;
+}) => ({
+  children: (
+    <ButtonElement variant="sort" className={TABLE_HEADER_BUTTON_CLASS_NAME}>
+      {capitalize(key)}
+      <IconElement
+        variant={
+          selection === key && direction !== 'none'
+            ? `sort-${direction}`
+            : 'sort-indeterminate'
+        }
+      />
+    </ButtonElement>
+  ),
+  key: `th=${key}`,
+  className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--${key}`,
+  onClick: () => onTableHeaderClick(key),
+  'aria-sort': selection === key ? direction : 'none',
+});
+
+const columnsArray = ['scope', 'type', 'permission'];
```

Comment:
still trying to figure this part out. At the moment need to have a custom display name